soh/soh/OTRGlobals.cpp

```diff
@@ -1793,6 +1794,78 @@ extern "C" int GetEquipNowMessage(char* buffer, char* src, const int maxBufferSi
     return 0;
 }
 
+extern "C" u8 GetNextChildTradeItem(u8 forward) {
+    std::vector<u8> possibleItems;
+
+    // We want this separate from the zelda's letter checks because technically they can
+    // get zelda's letter before waking talon up, which would have allowed them to get
+    // into a state where they cannot wake talon up, blocking the lon lon ranch checks
+    if (!Flags_GetEventChkInf(EVENTCHKINF_TALON_WOKEN_IN_CASTLE)) {
+        if (Flags_GetRandomizerInf(RAND_INF_WEIRD_EGG_HATCHED)) {
+            possibleItems.push_back(ITEM_CHICKEN);
+        // If weird egg hasn't hatched yet but been obtained
+        } else if (Flags_GetRandomizerInf(RAND_INF_OBTAINED_WEIRD_EGG)) {
+            possibleItems.push_back(ITEM_WEIRD_EGG);
+        }
+    }
+
+    if (!Flags_GetInfTable(INFTABLE_SHOWED_ZELDAS_LETTER_TO_GATE_GUARD)) {
+        if (Flags_GetEventChkInf(EVENTCHKINF_OBTAINED_ZELDAS_LETTER)) possibleItems.push_back(ITEM_LETTER_ZELDA);
+    }
+
+    if (CVarGetInteger("gMaskSelect", 0)) {
+        if (Flags_GetInfTable(INFTABLE_SHOWED_ZELDAS_LETTER_TO_GATE_GUARD)) {
+            if (Flags_GetItemGetInf(ITEMGETINF_OBTAINED_KEATON_MASK)) possibleItems.push_back(ITEM_MASK_KEATON);
+            if (Flags_GetItemGetInf(ITEMGETINF_OBTAINED_SKULL_MASK)) possibleItems.push_back(ITEM_MASK_SKULL);
+            if (Flags_GetItemGetInf(ITEMGETINF_OBTAINED_SPOOKY_MASK)) possibleItems.push_back(ITEM_MASK_SPOOKY);
+            if (Flags_GetItemGetInf(ITEMGETINF_OBTAINED_BUNNY_HOOD)) possibleItems.push_back(ITEM_MASK_BUNNY);
+            if (Flags_GetItemGetInf(ITEMGETINF_SOLD_ALL_MASKS)) {
+                possibleItems.push_back(ITEM_MASK_GORON);
+                possibleItems.push_back(ITEM_MASK_ZORA);
+                possibleItems.push_back(ITEM_MASK_GERUDO);
+                possibleItems.push_back(ITEM_MASK_TRUTH);
+            }
+        // Special case for bunny hood, if we want to start with it we don't care about happy mask shop status
+        } else if (Randomizer_GetSettingValue(RSK_STARTING_BUNNY_HOOD)) {
+            possibleItems.push_back(ITEM_MASK_BUNNY);
+        }
```

Comment:
maybe out of scope for this PR, but since start with bunny hood requires mask select to work we should probably either force it on (did we ever figure out a good pattern for forcing cvar options on based on rando settings?) or update the current cvar checks to use a macro or something that checks both the cvar and the RSK

Reply:
Starting with bunny hood doesn't require mask select, you could think of it like it just pretends you borrowed the bunny hood first.